backend/src/middlewares/medicationMiddlewares.js

```diff
@@ -31,3 +31,34 @@ async function deleteMedication(req, res, next) {
 }
 
 export { deleteMedication };
+
+async function postMedication(req, res, next) {
+  try {
+    const newMedication = await addNewMedication(req.body);
+    res.status(StatusCodes.CREATED).json(newMedication);
+  } catch (err) {
+    next(err);
+  }
+}
+
+export { postMedication };
+
+async function addNewMedication(medicationData) {
+  const existingMedication = await Medication.findOne({ nameOfMedication: medicationData.nameOfMedication });
```

Comment:
check for duplicates for single profile (we have many users with the same medication name)

Use medication.profile as an reference to Profile